### PR TITLE
Add initial tests to StoragePowerActor

### PR DESCRIPTION
**Pull request by zixuanzh**

### CODE REVIEW

File: actors/builtin/storage_power/storage_power_test.go [hunk 1]
```diff
@@ -1 +1,128 @@
...
+	"github.com/ipfs/go-cid"
+	"github.com/stretchr/testify/require"
+	"testing"
+
+	addr "github.com/filecoin-project/go-address"
+	builtin "github.com/filecoin-project/specs-actors/actors/builtin"
+	storage_power "github.com/filecoin-project/specs-actors/actors/builtin/storage_power"
+	adt "github.com/filecoin-project/specs-actors/actors/util/adt"
+	mock "github.com/filecoin-project/specs-actors/support/mock"
+	assert "github.com/stretchr/testify/assert"
```
Comment (anorth) [inline actors/builtin/storage_power/storage_power_test.go]: Please use goimports to sort and group arrays. GoLand has a facility to do this too (in settings)

File: actors/builtin/storage_power/storage_power_test.go [hunk 2]
```diff
@@ -1 +1,128 @@
...
+			SectorSize: abi.SectorSize(int64(32)),
+			Peer: "miner1",
+		}
+
+		rt := builder.Build(t)
+		actor.constructAndVerify(rt)
+
+		// owner1 send CreateMiner to StoragePowerActor
+		rt.SetCaller(owner1, builtin.AccountActorCodeID)
+		rt.SetReceived(big.Zero())
```
Comment (anorth) [inline actors/builtin/storage_power/storage_power_test.go]: Set a non-zero amount here and verify it gets accounted as pledge collateral  for the miner

File: actors/builtin/storage_power/storage_power_test.go [hunk 3]
```diff
@@ -1 +1,128 @@
...
+		rt.ExpectValidateCallerType(builtin.AccountActorCodeID, builtin.MultisigActorCodeID)
+
+		createMinerRet := &storage_power.CreateMinerReturn{
+			IDAddress:  miner1, // miner actor id address
+			RobustAddress: miner1, // should be long miner actor address
+		}
+		rt.ExpectSend(owner1, builtin.MethodSend, createMinerParams, abi.NewTokenAmount(10), &mock.ReturnWrapper{createMinerRet},0)
```
Comment (anorth) [inline actors/builtin/storage_power/storage_power_test.go]: Thanks, I can probably make this return wrapper more ergonomic later.

File: actors/builtin/storage_power/storage_power_test.go [hunk 4]
```diff
@@ -1 +1,128 @@
...
+		rt.Call(actor.StoragePowerActor.CreateMiner, createMinerParams)
+		rt.Verify()
+
+		var st storage_power.StoragePowerActorState
+		rt.GetState(&st)
+		assert.Equal(t, int64(1), st.MinerCount)
+		assert.Equal(t, abi.NewStoragePower(0), st.TotalNetworkPower)
+		assert.Equal(t, int64(0), st.NumMinersMeetingMinPower)
+		escrowTable := adt.AsMap(rt.Store(), st.ClaimedPower)
+		keys, err := escrowTable.CollectKeys()
```
Comment (anorth) [inline actors/builtin/storage_power/storage_power_test.go]: assert length of keys == 1. Then don't loop below, just directly do a `Get` for the miner1 address.

File: actors/builtin/storage_power/storage_power_test.go [hunk 5]
```diff
@@ -1 +1,128 @@
...
+		actor.constructAndVerify(rt)
+
+		// owner1 send CreateMiner to StoragePowerActor
+		rt.SetCaller(owner1, builtin.AccountActorCodeID)
+		rt.SetReceived(big.Zero())
+		rt.ExpectValidateCallerType(builtin.AccountActorCodeID, builtin.MultisigActorCodeID)
+
+		createMinerRet := &storage_power.CreateMinerReturn{
+			IDAddress:  miner1, // miner actor id address
+			RobustAddress: miner1, // should be long miner actor address
```
Comment (anorth) [inline actors/builtin/storage_power/storage_power_test.go]: Use addr.Undef for this second one - it should not be used by the power actor at all

File: actors/builtin/storage_power/storage_power_test.go [hunk 6]
```diff
@@ -1 +1,128 @@
...
+
+//
+// Misc. Utility Functions
+//
+
+type key string
+
+func asKey(in string) adt.Keyer {
+	return key(in)
+}
```
Comment (anorth) [inline actors/builtin/storage_power/storage_power_test.go]: You shouldn't need this key wrapper, the tables are keyed by address.

File: actors/builtin/storage_power/storage_power_test.go (hunk 3 above)
Comment (zixuanzh) [reply actors/builtin/storage_power/storage_power_test.go]: thanks 🙏

File: actors/builtin/storage_power/storage_power_test.go (hunk 5 above)
Comment (zixuanzh) [reply actors/builtin/storage_power/storage_power_test.go]: setting that to Undef will throw a cbor error

File: actors/builtin/storage_power/storage_power_test.go (hunk 6 above)
Comment (zixuanzh) [reply actors/builtin/storage_power/storage_power_test.go]: looks like we still need it tho

**Pull request merged**